commerce-subscription-web/src/main/java/com/liferay/commerce/subscription/web/internal/display/context/CPDefinitionSubscriptionInfoDisplayContext.java

```diff
@@ -85,6 +99,48 @@ public PortletURL getPortletURL() throws PortalException {
 		return portletURL;
 	}
 
+	public boolean hasRecurringPaymentMethod() throws PortalException {
+		List<CommerceChannelRel> commerceChannelRels =
+			CommerceChannelRelLocalServiceUtil.getCommerceChannelRels(
+				CPDefinition.class.getName(), getCPDefinitionId(), -1, -1,
+				null);
+
+		for (CommerceChannelRel commerceChannelRel : commerceChannelRels) {
+			boolean channelHasRecurringPaymentMethod = false;
+
+			CommerceChannel commerceChannel =
+				commerceChannelRel.getCommerceChannel();
+
+			List<CommercePaymentMethodGroupRel> commercePaymentMethodGroupRels =
+				_commercePaymentMethodGroupRelLocalService.
+					getCommercePaymentMethodGroupRels(
+						commerceChannel.getSiteGroupId(), true);
+
+			for (CommercePaymentMethodGroupRel commercePaymentMethodGroupRel :
+					commercePaymentMethodGroupRels) {
+
+				if (commercePaymentMethodGroupRel.isActive()) {
+					CommercePaymentMethod commercePaymentMethod =
+						_commercePaymentMethodRegistry.getCommercePaymentMethod(
+							commercePaymentMethodGroupRel.getEngineKey());
```

Comment:
also, this again seems a bit problematic to me. Doing IO inside a nested loop is a recipe for performance issues....

Reply:
Yeah, I'll have to talk to @marco-leo about it when I talk to him about the other stuff